core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/variables/runtime/VariableContext.scala

```diff
@@ -0,0 +1,111 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.accesscontrol.blocks.variables.runtime
+
+import cats.data.NonEmptyList
+import tech.beshu.ror.accesscontrol.blocks.rules.Rule
+import tech.beshu.ror.accesscontrol.blocks.variables.runtime.MultiExtractable.SingleExtractableWrapper
+
+object VariableContext {
+
+  sealed trait VariableType
```

Comment:
Do we have to introduce VariableType? Maybe we can use just Extractable to do the same? 
eg. SingleExtractable.UserIdVar & MultiExtractable.UserIdVar could be treated as VariableType.User.

Reply:
I think introducing `VariableType` makes design and intention more explicit, because we define requirements for variables, not for extractables (there is for example `Const` extractable and it is not related to variables in any way). If I used extractables I would need to use `Extractable[_]` everywhere and when pattern matching think about single and multivariables like:
```scala 
case SingleExtractable.UserIdVar | MultiExtractable.UserIdVar =>
```
For example defining `Show` instances would be less readable. So `VariableType` is grouping extractables (single and multi) under set of available variables.